Source/Plugins/Processing/ProcessingFilters/ConvertColorToGrayScale.cpp

```diff
@@ -58,100 +58,77 @@
 #include "Processing/ProcessingConstants.h"
 #include "Processing/ProcessingVersion.h"
 
-/**
- * @brief The ConvertColorToGrayScaleImpl class implements a threaded algorithm
- * that flattens an RGB array into a grayscale array
- */
-class LuminosityImpl
+struct Functor
 {
-public:
-  LuminosityImpl(uint8_t* data, uint8_t* newdata, FloatVec3_t colorWeights, size_t comp)
-  : m_ImageData(data)
-  , m_FlatImageData(newdata)
-  , m_ColorWeights(colorWeights)
-  , m_NumComp(comp)
-  {
-  }
-  virtual ~LuminosityImpl() = default;
+protected:
+  const size_t m_start{0};
```

Comment:
The {} syntax is used for initialization and prevents narrowing.  If the type does not match what it is being put into, it will not compile as it will not cast the value by default like `=`.  For instance, you cannot pass an int into a double.  cppreference has it as part of their list initialization example where they assign ints, doubles, and unsigned char without being arrays or vectors.

I agree with the suggested change, but I wanted to clarify the meaning of the other way.